x-pack/plugins/alerting/server/task_runner/types.ts

```diff
@@ -48,18 +48,22 @@ export interface RuleTaskInstance extends ConcreteTaskInstance {
   state: RuleTaskState;
 }
 
-export interface ScheduleActionsForRecoveredAlertsParams<
+export interface ScheduleActionsForAlertsParams<
   InstanceState extends AlertInstanceState,
   InstanceContext extends AlertInstanceContext,
+  ActionGroupIds extends string,
   RecoveryActionGroupId extends string
 > {
   logger: Logger;
   recoveryActionGroup: ActionGroup<RecoveryActionGroupId>;
   recoveredAlerts: Record<string, Alert<InstanceState, InstanceContext, RecoveryActionGroupId>>;
-  executionHandler: ExecutionHandler<RecoveryActionGroupId>;
+  executionHandler: ExecutionHandler<ActionGroupIds | RecoveryActionGroupId>;
```

Comment:
While i was working on the alerts summary POC i realised that having a separate RecoveryActionGroupId is making refactoring hard. It is forcing us to do tons of type conversion tricks. Can't we get rid of it? it's just another  ActionGroupId, why do we separate it?

Reply:
i mean, maybe we can create another issue under task_runner refactoring meta issue to remove RecoveryActionGroupId, not within this PR :)